src/run_throttlebot.py

```diff
@@ -1325,6 +1326,51 @@ def filter_mr(mr_allocation, acceptable_resources, acceptable_services, acceptab
 
     return mr_allocation
 
+def set_mr_provision_detect_id_change(redis_db, mr, new_mr_allocation, workload_config):
+    while True:
```

Comment:
Bad practice. Put a limit (say, 10) to the number of attempts here and terminate after those attempts. Also insert some sleeps (like every 10 seconds). At a certain point, there is a failure. At that point, just exit gracefully..

Can you throw back the logging at some point?